content-security-policy/webrtc/webrtc.js

```diff
@@ -0,0 +1,60 @@
+
+// Creates two RTCPeerConnection and tries to connect them. Returns
+// "allowed" if the connection is permitted, "blocked" if it is
+// blocked on both sides and "inconsistent" in the event that the
+// result is not the same on both sides (should never happen).
+async function tryConnect() {
+    const pc1 = new RTCPeerConnection();
+    const pc2 = new RTCPeerConnection();
+
+    // Returns a promise which resolves to a boolean which is true
+    // if and only if pc.iceConnectionState settles in the "failed"
+    // state, and never transitions to any state other than "new"
+    // or "failed."
+    const pcFailed = (pc) => {
+        return new Promise((resolve, _reject) => {
+            pc.oniceconnectionstatechange = (e) => {
+                if(pc.iceGatheringState === "complete") {
+                    resolve(pc.iceConnectionState === "failed");
+                } else if(pc.iceConnectionState !== "new") {
+                    resolve(false);
+                }
```

Comment:
The following should suffice, and align better with the code comment above (since `"new"` is the initial state):
```suggestion
                resolve(pc.iceConnectionState == "failed");
```